File: android/src/main/kotlin/site/shasmatic/flutter_veepoo_sdk/exceptions/VeepooException.kt
package site.shasmatic.flutter_veepoo_sdk.exceptions

/**
 * Base class for all exceptions thrown by the Veepoo SDK.
 *
 * @constructor Creates a new [VeepooException] with the given error message and cause.
 * @param message The error message.
 * @param cause The cause of the exception.
 * @see Exception
 * @see Throwable
 * @author Ahmad Rifa'i
 */
abstract class VeepooException(message: String, cause: Throwable? = null) : Exception(message, cause)

File: lib/src/exceptions/veepoo_exception.dart
part of '../../flutter_veepoo_sdk.dart';

/// {@template flutter_veepoo_sdk.veepoo_exception}
/// Base class for all exceptions thrown by the Veepoo package.
/// {@endtemplate}
final class VeepooException implements Exception {
  /// {@macro flutter_veepoo_sdk.veepoo_exception}
  const VeepooException({this.message, this.details, this.stacktrace});

  /// The exception message.
  final String? message;

  /// The exception details.
  final String? details;

  /// The stack trace for the exception.
  final StackTrace? stacktrace;

  @override
  String toString() {
    return 'VeepooException($message, $details, $stacktrace)';
  }
}
